esp/esp/web/util/latex.py

```diff
@@ -107,18 +109,24 @@ def gen_latex(texcode, type='pdf', remove_files=False, stdout=PIPE, stderr=STDOU
     :rtype:
         HttpResponse
     """
+    files = [stdout, stderr]
+    with nested(*(open(os.devnull, 'w') for f in files if f is devnull_sentinel)) as devnull_files:
+        for devnull_file in devnull_files:
+            files[files.index(devnull_sentinel)] = devnull_file
```

Comment:
Also, I don't think this will work correctly if you actually have `stdout==stderr==devnull_sentinel`.